### DISCUSSION TITLE
OpenDTU v24.8.5 not talking to Hoymiles HMS-2000-4T inverter

### DISCUSSION THREAD

**Discussion by the49ers2** [tbnobody/OpenDTU#2203] [Q&A] [answered]
Hi,

OpenDTU v24.8.5 not talking to Hoymiles HMS-2000-4T inverter

I have purchased a complete OpenDTU Unit on Ebay which contains:
- ESP32_Devkitc_V4 board with ESP32-WR00M-32U
- NRF24L01+ Transmisson module
- SSH1106 1,3" IIC Display
- two antennas

OpenDTUonBattery "was" installed (don't have the version), where it was not possible to update to the latest version, 
actually the system would not start after a 2nd update attempt (after several power cycles) and was dead (black display and not pingable).

I fixed the issue by connecting the device to a Windows computer and installed "CP210x_Universal_Windows_Driver.zip", so that it will show with a COM port within the Windows system.
I went to the site https://www.opendtu.solar/firmware/webinstall/ and installed ESP32 (v24.4.24) which worked without issues. 
Then I updated to the latest release (  v24.8.5 opendtu-generic_esp32.bin )
I was able do the network config and also got the SSH1106 display and NRF24L01+ back into the device manager by creating and uploading a self created pin_mapping.json file
------[pin_mapping.json Start]---------------------------------------------------
[
 {  
  "name": "Display SSH1106 and Radio NRF24L01+",
          "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "display": {
            "type": 3,
            "data": 21,
            "clk": 22
        }
  }
]
---------------[pin_mapping.json End]------------------------------------
The nRF24 Status is [configured], 
the nRF23 Chip status is [connected] 
and the nRF24 Chip Type is [nRF24L01+]
The Display is showing it’s data.

In inverter settings, I enter the Serial number 1161aXXXXXXX, OpenDTU detects the inverter as “HMS-1600/1800/2000-4T”, and a green light is constantly flashing besides “Live Data” on the main screen, but no data.
If I go to “show / set inverter limit”, (with “Last Limit Set Status: [OK]”) and set limit to 40% confirmed by [set limit persistent], then return to the “show / set inverter limit”, the “last limit set status” is [pending].
“Show Inverter Info” will display “No Information available, Did not receive any valid data from the inverter till now. Still trying...”

This all this happens, even if I place the Adapter right below the inverter.
I also set the "NRF24 Transmitting power" to Maximm (0 dBm), without noticing a change.
Searching thru the discussions, I found this German discussion https://github.com/tbnobody/OpenDTU/discussions/2041 releated to the LED settings causing issues in that case, the LED settings for led0 and led1 are both set to -1 on my setup.

I previously had a Hoymiles DTU-Wlite-S, connected to the HMS-2000-4T without issues.
While testing with OpenDTU, I unplugged the DTU-Wlite-S and also disconnected / connected the inverter from/to the mains.

I wonder if there is possibly of a Hoymiles protocol change, or OpenDTU has not been running with a Hoymiles HMS-2000-4T inverter, or something is wrong with the ESP32 module setup.

Is there a way to debug this properly, 
when I run the web based virtual debug console in parallel with the upper config changes, I only get the output:
----[debugscreen start]------------------------------------
Websocket: [/livedata][3] connect
19:11:07.722 > Websocket: [/livedata][3] disconnect
19:14:11.505 > Websocket: [/livedata][4] connect
19:16:21.714 > Websocket: [/livedata][4] disconnect
19:16:24.520 > Websocket: [/livedata][5] connect
----[debugscreen end]--------------------------------------

Hope you have an idea how this can be fixed.

Best
Justin

**Comment by Kroki0815** [comment]
Hi,
the HMS/HMT-Series need the CMT2300A Chip for communication. Only the HM-Series works with the NRF24 Chip.

**Comment by stefan123t** [comment, marked as answer]
@the49ers2 you did well as you explained everything in detail and succeeded in getting the OpenDTU board running with NRF24L01+ as a Radio. 

Unfortunately the HMS-2000-4T will only work with a CMT2300A radio chip.
You may want to consider buying an OpenDTU Fusion board which has both radio chips integrated on board or buy a CMT2300A breakout board. The former has been tested extensively by the developers, it is our basically our test platform.

While the HMS-2000**W** will not work with OpenDTU at all having a built-in Hoymiles DTUBI which can only be queried via WiFi AP in Installer Mode or directly send its data to the Hoymiles cloud.

**Reply by the49ers2** [reply]
Thank you both stefan123t and Kroki0815 for the quick feedback, 

would have been good to have a note in the respective module sections
https://www.opendtu.solar/hardware/nrf24l01plus/ and
https://www.opendtu.solar/hardware/cmt2300a/
for which Hoymiles units these modules work (and do not work),
referring to the inverter overview list ( https://www.opendtu.solar/hardware/inverter_overview/), which lists the correct module with the respective inverter. 

It's now too late for me, will be getting a CMT2300A.

Best
justin

**Reply by tbnobody** [reply]
> would have been good to have a note in the respective module sections
> It's now too late for me, will be getting a CMT2300A.

are you blaming the manual for something you have overlooked?
https://www.opendtu.solar/hardware/inverter_overview/
